src/util-fix_checksum.c

```diff
@@ -54,7 +54,6 @@ FixChecksum(uint16_t sum, uint16_t old, uint16_t new)
 
     l = sum + old - new;
     l = (l >> 16) + (l & 65535);
-    l = l & 65535;
 
-    return l;
+    return (uint16_t)l;
```

Comment:
@jasonish any thoughts here? IIRC this func was introduced by you?

Reply:
This is an easy one, without knowing the function's purpose
`l = l & 65535; return l;` does the same thing as  `return (uint16_t)l;` (right ?)

Reply:
I guess I'm not actually sure what C is doing (and/or expected to do here), esp since this code IIRC comes from OpenBSD. They seem to know a thing or 2 about this stuff.

Reply:
> `l = l & 65535; return l;` does the same thing as `return (uint16_t)l;` (right ?)

A quick test shows that they do the same thing, but its not nearly as clear to me as the original form.

Reply:
Do you prefer `l = l & 65535; (uint16_t)return l;` ? (even if it is redundant)

Reply:
I think so, at least its clear.. Would this work as well?
```
return (uint16_t)(l & 65535);
```

Reply:
ok for me